test/versioning-strategies/prerelease.ts

```diff
@@ -0,0 +1,280 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {describe, it} from 'mocha';
+
+import {expect} from 'chai';
+import {PrereleaseVersioningStrategy} from '../../src/versioning-strategies/prerelease';
+import {Version} from '../../src/version';
+
+describe('PrereleaseVersioningStrategyVersioningStrategy', () => {
+  describe('with breaking change', () => {
+    const commits = [
+      {
+        sha: 'sha1',
+        message: 'feat: some feature',
+        files: ['path1/file1.txt'],
+        type: 'feat',
+        scope: null,
+        bareMessage: 'some feature',
+        notes: [],
+        references: [],
+        breaking: false,
+      },
+      {
+        sha: 'sha2',
+        message: 'fix!: some bugfix',
+        files: ['path1/file1.rb'],
+        type: 'fix',
+        scope: null,
+        bareMessage: 'some bugfix',
+        notes: [{title: 'BREAKING CHANGE', text: 'some bugfix'}],
+        references: [],
+        breaking: true,
+      },
+      {
+        sha: 'sha3',
+        message: 'docs: some documentation',
+        files: ['path1/file1.java'],
+        type: 'docs',
+        scope: null,
+        bareMessage: 'some documentation',
+        notes: [],
+        references: [],
+        breaking: false,
+      },
+    ];
+    it('can bump a major', async () => {
+      const strategy = new PrereleaseVersioningStrategy();
+      const oldVersion = Version.parse('1.2.3');
+      const newVersion = await strategy.bump(oldVersion, commits);
+      expect(newVersion.toString()).to.equal('2.0.0');
+    });
+
+    it('can bump a major on pre major for breaking change', async () => {
+      const strategy = new PrereleaseVersioningStrategy();
+      const oldVersion = Version.parse('0.1.2');
+      const newVersion = await strategy.bump(oldVersion, commits);
+      expect(newVersion.toString()).to.equal('1.0.0');
```

Comment:
We don't actually use "less than 0" majors, but I'd expect this to be 0.2.0 personally. (Basically "before 1.0, there are no guarantees.") I've now read that this is the "bumpMinorPreMajor" behavior, but I'd personally expect it to be the default. It feels very weird for "I've taken a breaking change" to be the trigger to go from 0.x to 1.0.